### PR TITLE
Crawler/init

### PR DESCRIPTION
**Pull request by rsekulski-scalac**
First version of crawler.
Missing common-schema.

### CODE REVIEW

File: crawler/src/main/scala/io/scalac/newspaper/crawler/Main.scala [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  that =>
+
+  implicit val system = ActorSystem("crawler-system")
+  implicit val ec = system.dispatcher
+  implicit val materializer = ActorMaterializer()
+  val config = ConfigFactory.load
+
+  val fetchingProcess = new FetchingProcess with FileURLsStore with HttpURLFetcher with HttpFetchingFlow with KafkaPublisher {
+    override implicit val ec: ExecutionContext = that.ec
+    override val system: ActorSystem = that.system
```
Comment (tues) [inline crawler/src/main/scala/io/scalac/newspaper/crawler/Main.scala]: This gets initialized before `that.system`, which results in a `NullPointerException` in `KafkaPublisher`.

File: crawler/src/main/resources/application.conf [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      EverydayMidnight {
+        description = "Everyday at midnight"
+        expression = "0 0 0 * * ?"
+      }
+    }
+  }
+}
+
+bootstrap {
+  servers = "localhost:9092"
```
Comment (tues) [inline crawler/src/main/resources/application.conf]: This should be `akka.kafka.producer.kafka-clients.bootstrap.servers`, not just `bootstrap.servers`.

File: crawler/src/main/resources/application.conf [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+crawler {
+  http {
+    timeout = 10s
+    max.parallelism = 30
+  }
+}
+
+kafka {
+  topic = "newspaper-content"
+}
```
Comment (tues) [inline crawler/src/main/resources/application.conf]: How about adding `include "environment.conf"` to make testing easier for Patryk? :)

File: crawler/build.sbt [hunk 4]
```diff
@@ -0,0 +1,43 @@
+name := "newspaper-crawler"
+
+version := "1.0"
+
+scalaVersion := "2.12.1"
```
Comment (pjazdzewski1990) [inline crawler/build.sbt]: Reactive kafka was causing issues with 2.12

File: crawler/src/main/resources/application.conf (hunk 3 above)
Comment (pjazdzewski1990) [inline crawler/src/main/resources/application.conf]: I can live without it - in worse case I will add it myself later ;) 
But in general could it be useful - we won't wait till midnight to wait for pages to be fetched :P

File: crawler/src/main/scala/io/scalac/newspaper/crawler/publishing/Serialization.scala [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+      override def close(): Unit = ()
+
+      override def deserialize(topic: String, data: Array[Byte]): T = {
+        messageCompanion.parseFrom(data)
+      }
+
+    }
+
+class ContentFetchedSerializer extends ProtobufSerializer(ContentFetched)
+class ContentFetchedDeserializer extends ProtobufDeserializer(ContentFetched)
```
Comment (pjazdzewski1990) [inline crawler/src/main/scala/io/scalac/newspaper/crawler/publishing/Serialization.scala]: Do we need both serializer and deserializer?

File: crawler/src/main/scala/io/scalac/newspaper/crawler/urls/HardcodedURLsStore.scala [hunk 6]
```diff
@@ -0,0 +1,15 @@
+package io.scalac.newspaper.crawler.urls
+
+import akka.NotUsed
+import akka.stream.scaladsl.Source
+import scala.collection.immutable.Seq
+
+trait HardcodedURLsStore extends URLsStore {
```
Comment (pjazdzewski1990) [inline crawler/src/main/scala/io/scalac/newspaper/crawler/urls/HardcodedURLsStore.scala]: Is this used?

File: crawler/build.sbt (hunk 4 above)
Comment (pjazdzewski1990) [inline crawler/build.sbt]: Ideally the version should not matter. Who knows ... maybe we will write one service in PHP ...

Comment (tues) [inline crawler/build.sbt]: Yes, but being able to jump to definitions is very handy when doing code review, so I needed to switch to unstable ENSIME 2.x. Well, one step closer to contributing to that project ;)

In the worst case I can easily work without IDE-like features, I've grown up without them.

So from my perspective it doesn't really matter much. Just pointing out that if we're going to use `2.12`, then we can use `2.12.2` already, no need for `2.12.1`. I'm staying with `2.11.11` for now.

File: crawler/src/main/resources/application.conf (hunk 3 above)
Comment (tues) [inline crawler/src/main/resources/application.conf]: I think in the long run we shouldn't be using cron-like scheduling, but fetching pages in fixed intervals distributed throughout the day. Until then yes, I had to change that line to `0 * * * * ?` to test if it works, so `environment.conf` would be double useful.

File: crawler/build.sbt (hunk 4 above)
Comment (rsekulski-scalac) [inline crawler/build.sbt]: Yes, looks like akka-quartz-scheduler for akka 2.5.x is only available for scala 2.12.x.
I'll switch to 2.12.2.

File: crawler/src/main/scala/io/scalac/newspaper/crawler/publishing/Serialization.scala (hunk 5 above)
Comment (rsekulski-scalac) [inline crawler/src/main/scala/io/scalac/newspaper/crawler/publishing/Serialization.scala]: No, but I'll remove it anyway after merging with common-schema

File: crawler/src/main/resources/application.conf [hunk 7]
```diff
@@ -0,0 +1,37 @@
+akka {
+  quartz {
```
Comment (tues) [inline crawler/src/main/resources/application.conf]: This is no longer used, right?

**Pull request merged**